app/helpers/application_helper/button/middleware_domain_server_action.rb

```diff
@@ -8,6 +7,6 @@ def visible?
   private
 
   def in_domain?
-    @record.try(:in_domain?) || @record.try(:middleware_server).try(:in_domain?)
+    !@record.nil? && (@record.try(:in_domain?) || @record.try(:middleware_server).try(:in_domain?))
```

Comment:
I don't understand why you removed the `need :@record` this line seems to be doing that test.

Also, please, use `.present?` rather than `!.nil?`.